crates/telnet-host/src/telnet.rs

```diff
@@ -175,19 +187,72 @@ impl TelnetConnection {
                     };
                     let line = line.unwrap();
 
-                    // Are we expecting to respond to prompt input? If so, send this through to that.
-                    let response = match expecting_input_reply.take() {
-                        Some(input_reply_id) => {
-                            rpc_client.make_rpc_call(self.client_id, RpcRequest::RequestedInput(self.client_token.clone(), auth_token.clone(), input_reply_id, line)).await?
-                        }
-                        None => {
+                    let response = match line_mode.clone() {
+                        LineMode::Input => {
+                            // If the line is .program <verb> ... then we need to start spooling up a program.
+                            // But we do need to do some very basic parsing to get the target and verb and reject complete nonsense.
+                            // Note that LambdaMOO is more fussy and the server validates the object and verb etc. before accepting the program.
+                            if line.starts_with(".program") {
```

Comment:
Once in this state machine, I think you need to support .flush on a newline. - it's intended to purge the buffer.

Reply:
Good point. I should add that. It's funny I went for years back then as a MOO user without even knowing that .flush existed.

Reply:
Actually, so, I misunderstood `.flush`.  I'm not sure we want/need that.

```
It sometimes happens that a user changes their mind about having typed one or more lines of input and would like to `untype' them before the server actually gets around to processing them. If they react quickly enough, they can type their connection's defined flush command; when the server first reads that command from the network, it immediately and completely flushes any as-yet unprocessed input from that user, printing a message to the user describing just which lines of input were discarded, if any.
```

This is a bit crazy, relies on the user getting .flush in before the task queue can process things. Which, well, on a modern system is highly unlikely and probably only really worked for LambdaMOO back in the day because it was stupid slow.

So, doesn't have any specific relevance in the context of `.program` (I thought maybe it turfed the spooled up program buffer but nope)

```
flush
>> No pending input to flush...
.program me:test
Now programming Karl:test.  Use "." to end.
.flush
>> No pending input to flush...
5;
.flush
>> No pending input to flush...
.
0 error(s).
Verb programmed.
```